### ISSUE TITLE
Importing Html reports from Local to  Report Portal

### ISSUE THREAD

**Issue by DEEPAKCHANDRA7** [reportportal/reportportal#1334]
Hi @DzmitryHumianiuk 

 

We are trying to import HTML report from Local that are created while running automation scripts to Report Portal dashboard for that specific launch. Please let me know how to do this Thanks ! 

 

PS:Html reports also contains screenshots(.jpg files) that took using Phyton !
Can you please help me with this?

**Comment by DzmitryHumianiuk** [comment]
@DEEPAKCHANDRA7 there is possibility to upload attachments as Launch Logs.
you will see button `Log` once go inside launch (see pic).
RP can receive single files. But it won't represent in a right way a structure of html+pictures.

In java client we have implemented method in client: https://github.com/reportportal/client-java/blob/develop/src/main/java/com/epam/reportportal/service/LaunchLoggingContext.java#L126
but i'm not sure about the same in python client. @iivanou is more familiar with it.


![image](https://user-images.githubusercontent.com/11332788/110532728-428dcc80-812e-11eb-9dd2-e808479c2511.png)

**Comment by iivanou** [comment]
I think this request is about adding test results to RP using an already created HTML file. For example, it may be an HTML file created by the allure plugin in Pytest. @DEEPAKCHANDRA7 Is that what you are trying to accomplish?

**Comment by jayanthktn** [comment]
I need that too. i have generated the custom report, i need to attach that to RP

**Comment by FelipeJK** [comment]
Is there a plan to support HTML files uploaded to Report Portal programmatically @iivanou?
I did some uploads using Spark report from Extent Reports and all HTML files open without loading the css properly.

![image](https://user-images.githubusercontent.com/18666206/128393962-34ed9624-b89e-47bb-88b1-72025831ac19.png)

**Comment by DzmitryHumianiuk** [comment]
@FelipeJK are we talking about importing the results of the entire launch? or about attaching a single HTML file in the test logs?

**Comment by realqpat** [comment]
@DzmitryHumianiuk I'm having the same issue reported (attaching a single HTML to ta launch)
Any idea when this will be fixed ?

**Comment by DzmitryHumianiuk** [comment]
@realqpat could you please give more details?
1. is this the issue with a programmatic implementation, and you can't send the HTML file as a launch attachment?
2. is it sent, but not visible in RP pages?
3. is it sent, visible in RP pages, but rendered incorrectly?

**Comment by FelipeJK** [comment]
> @realqpat could you please give more details?
> 
> 1. is this the issue with a programmatic implementation, and you can't send the HTML file as a launch attachment?
> 2. is it sent, but not visible in RP pages?
> 3. is it sent, visible in RP pages, but rendered incorrectly?

Option 3. HTML files are not rendered correctly

**Comment by DzmitryHumianiuk** [comment]
@FelipeJK does it look good if you download it locally?
i mean, if the HTML file has the reference to the CSS files and etc., it won't work as expected.

**Comment by FelipeJK** [comment]
If I download locally it will open fine. 8 changed my code to upload a zip file. It helps when css is in a different folder.

**Comment by realqpat** [comment]
Correct.. when I open the HTML within RP (open in new tab), it's missing the css. But if I download it, it's displayed properly
We are migrating to RP and our current tool allow user to simply view the HTML (without resorting to download it)

**Comment by YerassylNurzhan** [comment]
The bug is reproducible and added to backlog

**Comment by Bam6ycha** [comment]
> Correct.. when I open the HTML within RP (open in new tab), it's missing the css. But if I download it, it's displayed properly We are migrating to RP and our current tool allow user to simply view the HTML (without resorting to download it)

Good day.
I have several questions to you.
The first one is: Can you tell witch tool do you use?
And the second one is : You send as attachment .html file witch you  maybe have  locally?
Thanks for your answer.
